bpe.go

```diff
@@ -0,0 +1,110 @@
+package bpe
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+
+	"github.com/sirupsen/logrus"
+)
+
+type TokenId int32
```

Comment:
```suggestion
type TokenId uint32
```
this is what they have in C++ I think